js/components/thermoscope.js

```diff
@@ -10,18 +10,22 @@ import '../../css/thermoscope.less';
 
 const MODEL_WIDTH = 400;
 const MODEL_HEIGHT = 400;
+var sensor;
 
 export default class Thermoscope extends PureComponent {
   constructor(props) {
     super(props);
     this.state = {
       temperature: 20,
+      liveData: false,
       materialType: 'solid',
       materialIdx: 0
     };
     this.handleTempSliderChange = this.handleTempSliderChange.bind(this);
     this.handleMaterialTypeChange = this.handleMaterialTypeChange.bind(this);
     this.handleMaterialIdxChange = this.handleMaterialIdxChange.bind(this);
+    this.sensor = this.props.sensor;
```

Comment:
I know that it's gonna work in our specific case, but general React comment:
Probably it's not the best idea to assign property to some variable in the constructor. Property can be changed later, so usually it makes sense to always use `this.props.<propertyName>`.